components/combobox/combobox.jsx

```diff
@@ -362,6 +388,10 @@ class Combobox extends React.Component {
 			this.setState({ isOpen: nextProps.isOpen });
 		}
 
+		if (this.props.errorText) {
+			this.generatedErrorId = shortid.generate();
```

Comment:
I like what you've done. Let's simplify and generate this just on mount always.

Reply:
I have it on willRecieveProps because when I have it onMount only, the id's don't show up.

Reply:
Not sure why, since when the props change doesn't the component re-mount?

Reply:
It should if you remove `this.props.errorText`

Reply:
It should if you remove `if (this.props.errorText) {`

Reply:
Mounting only occurs once. Prop change doesn't trigger mount. This is the lifecyle diagram for React 15:

![1_u8hTumGAPQMYZIvfgQMfPA](https://user-images.githubusercontent.com/1290832/55261638-10308080-5231-11e9-881d-e7cc0c785886.jpg)

Reply:
Ohh I understand now. Thanks!